src/ASM/LR/ASMu2Dnurbs.C

```diff
@@ -0,0 +1,503 @@
+    double w = lrspline->getBasisfunction(MNPC[iel][i])->cp(wi);
+    W += tmp.basisValues[i]*w;
+    Wderxi += tmp.basisDerivs_u[i]*w;
+    Wdereta += tmp.basisDerivs_v[i]*w;
+    Wder2xi += tmp.basisDerivs_uu[i]*w;
+    Wder2eta += tmp.basisDerivs_vv[i]*w;
+    Wderxieta += tmp.basisDerivs_uv[i]*w;
+  }
+
+  bas.prepareDerivs(u, v, 0, -1, tmp.basisValues.size());
+  for (size_t i = 0; i < tmp.basisValues.size(); ++i) {
+    double w = lrspline->getBasisfunction(MNPC[iel][i])->cp(wi);
+    bas.basisValues[i] = tmp.basisValues[i]*w/W;
+    double H1 = (tmp.basisDerivs_u[i]*W-tmp.basisValues[i]*Wderxi);
+    bas.basisDerivs_u[i] = H1*w/(W*W);
+    double H2 = (tmp.basisDerivs_v[i]*W-tmp.basisValues[i]*Wdereta);
+    bas.basisDerivs_v[i] = H2*w/(W*W);
+    double dH1dx = tmp.basisDerivs_uu[i]*W-tmp.basisValues[i]*Wder2xi;
+    bas.basisDerivs_uu[i] = (dH1dx*W-2*H1*Wderxi)*w/(W*W*W);
+    double dH2dy = tmp.basisDerivs_vv[i]*W-tmp.basisValues[i]*Wder2eta;
+    bas.basisDerivs_vv[i] = (dH2dy*W-2*H2*Wdereta)*w/(W*W*W);
+    double dH1dy = tmp.basisDerivs_uv[i]*W + tmp.basisDerivs_u[i]*Wdereta -
+                   tmp.basisDerivs_v[i]*Wderxi - tmp.basisValues[i]*Wderxieta;
+    bas.basisDerivs_uv[i] = (dH1dy*W-2*H1*Wdereta)*w/(W*W*W);
+  }
+}
+
+
+void ASMu2Dnurbs::computeBasis(double u, double v,
+                               Go::BasisDerivsSf3& bas, int iel) const
+{
+  if (noNurbs)
+    return this->ASMu2D::computeBasis(u,v,bas,iel);
+
+  Go::BasisDerivsSf3 tmp;
+  lrspline->computeBasis(u,v,tmp,iel);
+  size_t wi = lrspline->getBasisfunction(MNPC[iel][0])->dim()-1;
+
+  double W = 0.0;
+  double Wderxi = 0.0;
+  double Wdereta = 0.0;
+  double Wder2xi = 0.0;
+  double Wder2eta = 0.0;
+  double Wderxieta = 0.0;
+  double Wder3xi = 0.0;
+  double Wder3eta = 0.0;
+  double Wder2xieta = 0.0;
+  double Wderxi2eta = 0.0;
+  for (size_t i = 0; i < tmp.basisValues.size(); ++i) {
+    double w = lrspline->getBasisfunction(MNPC[iel][i])->cp(wi);
+    W += tmp.basisValues[i]*w;
+    Wderxi += tmp.basisDerivs_u[i]*w;
+    Wdereta += tmp.basisDerivs_v[i]*w;
+    Wder2xi += tmp.basisDerivs_uu[i]*w;
+    Wder2eta += tmp.basisDerivs_vv[i]*w;
+    Wderxieta += tmp.basisDerivs_uv[i]*w;
+    Wder3xi += tmp.basisDerivs_uuu[i]*w;
+    Wder3eta += tmp.basisDerivs_vvv[i]*w;
+    Wder2xieta = tmp.basisDerivs_uuv[i]*w;
+    Wderxi2eta = tmp.basisDerivs_uvv[i]*w;
+  }
+
+  bas.prepareDerivs(u, v, 0, -1, tmp.basisValues.size());
+  for (size_t i = 0; i < tmp.basisValues.size(); ++i) {
+    double w = lrspline->getBasisfunction(MNPC[iel][i])->cp(wi);
+    bas.basisValues[i] = tmp.basisValues[i]*w/W;
+    double H1 = (tmp.basisDerivs_u[i]*W-tmp.basisValues[i]*Wderxi);
+    bas.basisDerivs_u[i] = H1*w/(W*W);
+    double H2 = (tmp.basisDerivs_v[i]*W-tmp.basisValues[i]*Wdereta);
+    bas.basisDerivs_v[i] = H2*w/(W*W);
+    double dH1dx = tmp.basisDerivs_uu[i]*W-tmp.basisValues[i]*Wder2xi;
+    double dH2dx = tmp.basisDerivs_vv[i]*W-tmp.basisValues[i]*Wder2eta;
```

Comment:
This can't be right. `dH2dx` is here the same as `dH2dy` in l433.
Shouldn't it be equal to `dH1dy`instead (symmetry)?